docs/src/main/asciidoc/security-openid-connect-multitenancy-concept.adoc

```diff
@@ -0,0 +1,317 @@
+====
+
+
+The only required method to implement is `Uni<OidcTenantConfig> resolve(RoutingContext routingContext, OidcRequestContext<OidcTenantConfig> requestContext);`. If possible, cache realm IDs and their corresponding `OidcTenantConfig` at application launch to yield good performance during application runtime. A simple structure for resolving tenant configuration is shown below.
+
+. The resolve method considers the request path retrieved from the RoutingContext. This approach yields scalable performance.
+. In the resolve method, the `tenant identifier` is named `realm`
+. The `OidcTenantConfig` factory itself is not shown in the code snippet below as it focuses on the tenant configuration resolution part.
+. The `keycloak.url` configuration property would normally be qualified per environment, enabling different keycloak installations in development, test, staging and production environments.
+
+[source,kotlin]
+----
+@Singleton
+@AddDefaultConstructor
+@UseOpenMembers
+class OidcConfigurationResolver @Inject constructor(
+
+  @ConfigProperty(name = "keycloak.url")
+  val oidcServiceBaseURL: String
+
+) : TenantConfigResolver {
+
+  @PostConstruct
+  fun postConstruct() {
+
+    log.info("Using oidcServiceBaseURL [$oidcServiceBaseURL]")
+
+    // Populate the REALM_2_OIDC_CONFIG cache
+  }
+
+  override fun resolve(routingContext: RoutingContext?,
+                       requestContext: OidcRequestContext<OidcTenantConfig>?): Uni<OidcTenantConfig>? {
+
+    // #0) Check sanity
+    //
+    if (routingContext == null) {
+
+      log.warn("Received null RoutingContext. Cannot determine realm; reverting to default tenant.")
+      return Uni.createFrom().nullItem()
+    }
+
+    // #1) Find the tenant identifier/realm, from the path.
+    //
+    val contextPath = routingContext.request().path()
+    val realm = getRealmFrom(contextPath)
+
+    if (realm == null) {
+
+      log.warn("== No Realm derived from request path [$contextPath]")
+      return Uni.createFrom().nullItem()
+    } else {
+      log.debug("== Realm [$realm] found from request path [$contextPath]")
+    }
+
+    // #2) Fallback to null result if the tenant identifier was not recognized
+    //
+    return when (val config = REALM_2_OIDC_CONFIG[realm]) {
+      null -> Uni.createFrom().nullItem()
+      else -> Uni.createFrom().item(config)
+    }
+  }
+
+  companion object {
+
+    @JvmStatic
+    internal val REALM_2_OIDC_CONFIG: SortedMap<String, Supplier<OidcTenantConfig>> = TreeMap()
+
+    @JvmStatic
+    private val log = LoggerFactory.getLogger(OidcConfigurationResolver::class.java)
+  }
+}
+----
+
+This implementation concludes steps `1` in Figure 2 above. Quarkus is now aware of the tenant identifier/realm, and has resolved the configuration for the OIDC provider which authenticated the calling user.
+
+=== 2. Retrieve injected IdToken and user information
```

Comment:
This is an interesting summary that follows below, but I think it is off-topic for the multi-tenancy concept you introduce here, things like how to access the tokens, difference between them, etc. I'd like to propose for you to have a look at the current OIDC Bearer and Code flow tokens and reapply some of your proposed text below, does it sound reasonable ? Here it gets a bit technical while the main purpose is to explain the multi-tenancy concept